src/widgets/ot_color_picker.eliom

```diff
@@ -109,53 +110,44 @@ let display_sl_grid ~setter ((sel_hue, sel_sat, sel_ltn) as sel) =
         let el =
           D.(
             div
-              ~a:[
-                a_class (
-                  if is_selected
-                  then ["ot-color-picker-sl-picker-cell"; "ot-color-picker-selected-cell"]
-                  else ["ot-color-picker-sl-picker-cell"]);
-                a_style style';
-                a_onmousedown
-                  [%client fun _ ->
-                    ~%setter (~%sel_hue, ~%saturation, ~%lightness)
-                  ]
-              ]
-              []
-          )
+              ~a:
+                [
+                  a_class
+                    ( if is_selected then
+                      [
+                        "ot-color-picker-sl-picker-cell";
+                        "ot-color-picker-selected-cell";
+                      ]
+                    else [ "ot-color-picker-sl-picker-cell" ] );
+                  a_style style';
+                  a_onmousedown
+                    [%client
+                      fun _ -> ~%setter (~%sel_hue, ~%saturation, ~%lightness)];
+                ]
+              [])
         in
-        rows.(row) <- el :: rows.(row)
-      )
+        rows.(row) <- el :: rows.(row))
       irange
   in
   let rows =
     Array.map
-      (fun r ->
-        D.(
-          div
-            ~a:[a_class ["ot-color-picker-sl-picker-row"]]
-            r))
+      (fun r -> D.(div ~a:[ a_class [ "ot-color-picker-sl-picker-row" ] ] r))
       rows
   in
-  D.(
-    div
-      ~a:[a_class ["ot-color-picker-sl-picker"]]
-      (Array.to_list rows)
-  )
+  D.(div ~a:[ a_class [ "ot-color-picker-sl-picker" ] ] (Array.to_list rows))
 
-let display_aux ?(a=[]) ~setter ((sel_hue, sel_sat, sel_ltn) as sel) =
+let display_aux ?(a = []) ~setter ((sel_hue, sel_sat, sel_ltn) as sel) =
   let open Eliom_content.Html in
-    D.div ~a:(D.a_class ["ot-color-picker"]::a)
-      [ display_hue_selector ~setter sel
-      ; display_sl_grid ~setter sel
-      ]
+  D.div
+    ~a:(D.a_class [ "ot-color-picker" ] :: a)
+    [ display_hue_selector ~setter sel; display_sl_grid ~setter sel ]
 
 let display ?a cp_sig =
   let setter = snd cp_sig in
-  fst cp_sig >|= [%shared display_aux ?a:~%a ~setter:~%setter] |> Eliom_content.Html.R.node
+  fst cp_sig
+  >|= [%shared display_aux ?a:~%a ~setter:~%setter]
+  |> Eliom_content.Html.R.node
 
 let make ?a =
   let cp_sig = Eliom_shared.React.S.create (255, 1.0, 0.0) in
-  ( display ?a
-      cp_sig
-  , fst cp_sig
-  )
+  (display ?a cp_sig, cp_sig)
```

Comment:
I think the best option would be, rather than *always* creating a new signal when making this widget, to have it take an optional reactive component (setter + signal tuple).

If one is given, use that one, if not, create a new one and use it.
something like `let make ?a ?value_r () = ....`

Reply:
We should probably switch most of the widgets to this behavior, actually, as it allows for more user-friendly usage of them